docarray/array/storage/weaviate/backend.py

```diff
@@ -0,0 +1,290 @@
+        """
+        if not self._meta_name:
+            raise ValueError('meta object is not defined')
+
+        resp = (
+            self._client.query.get(self._meta_name, ['_offset2ids', '_additional {id}'])
+            .do()
+            .get('data', {})
+            .get('Get', {})
+            .get(self._meta_name, [])
+        )
+
+        if not resp:
+            return [], None
+        elif len(resp) == 1:
+            return resp[0]['_offset2ids'], resp[0]['_additional']['id']
+        else:
+            raise ValueError('received multiple meta copies which is invalid')
+
+    @property
+    def array_id(self):
+        """An alias to _class_name that returns the id/name of the class
+        in the weaviate of this :class:`DocumentArrayWeaviate`
+
+        :return: name of weaviate class/schema of this :class:`DocumentArrayWeaviate`
+        """
+        return self._class_name
+
+    @property
+    def _class_name(self):
+        """Return the name of the class in weaviate of this :class:`DocumentArrayWeaviate
+
+        :return: name of weaviate class/schema of this :class:`DocumentArrayWeaviate`
+        """
+        if not self._schemas:
+            return None
+        return self._schemas['classes'][0]['class']
+
+    @property
+    def _meta_name(self):
+        """Return the name of the class in weaviate that stores the meta information of
+        this :class:`DocumentArrayWeaviate`
+
+        :return: name of weaviate class/schema of class that stores the meta information
+        """
+        # TODO: remove this after we combine the meta info to the DocumentArray class
+        if not self._schemas:
+            return None
+        return self._schemas['classes'][1]['class']
+
+    @property
+    def _class_schema(self) -> Optional[Dict]:
+        """Return the schema dictionary of this :class:`DocumentArrayWeaviate`'s weaviate schema
+
+        :return: the dictionary representing this weaviate schema
+        """
+        if not self._schemas:
+            return None
+        return self._schemas['classes'][0]
+
+    @property
+    def _meta_schema(self):
+        """Return the schema dictionary of this weaviate schema that stores this object's meta
+
+        :return: the dictionary representing a meta object's weaviate schema
+        """
+        if not self._schemas and len(self._schemas) < 2:
+            return None
+        return self._schemas['classes'][1]
+
+    def _doc2weaviate_create_payload(self, value: 'Document'):
+        """Return the payload to store :class:`Document` into weaviate
+
+        :param value: document to create a payload for
+        :return: the payload dictionary
+        """
+        if value.embedding is None:
+            embedding = [0]
+        elif isinstance(value.embedding, scipy.sparse.spmatrix):
+            embedding = value.embedding.toarray()
+        else:
+            embedding = value.embedding
+
```

Comment:
to support `pytorch` tensor embedding

```suggestion
from ...math.ndarray import to_numpy_array
embedding = to_numpy_array(embedding)
```

Reply:
Addressed.